### PR TITLE
feat(2712): remove the accounting service interface from rafiki

### PR DESCRIPTION
**Pull request by koekiebox**
# Story:
https://github.com/interledger/rafiki/issues/2712

## Changes proposed in this pull request

- Remove the middleware `AccountService`

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [x] Documentation added
- [x] Make sure that all checks pass
- [x] Bruno collection updated

### CODE REVIEW

File: packages/backend/src/accounting/service.ts [hunk 1]
```diff
@@ -57,12 +57,16 @@ export interface Transaction {
   void: () => Promise<void | TransferError>
 }
 
-export interface AccountingService {
+export interface BasicAccountingService {
```
[the remark this answers is not in the document]
Comment (koekiebox) [reply packages/backend/src/accounting/service.ts]: @mkurapov  I had it like that before, and thought it was a bit nasty, but it is only used in the test case (Mock).

I updated it now. Ready for re-review.

File: packages/backend/src/payment-method/ilp/connector/core/rafiki.ts [hunk 2]
```diff
@@ -210,14 +187,6 @@ export class Rafiki<T = any> {
     return response.rawReply
   }
 
-  //public get router(): Router | undefined {
```
Comment (mkurapov) [inline packages/backend/src/payment-method/ilp/connector/core/rafiki.ts:213]: Thanks for removing these, I think there are just a few of these left in `/src/payment-method/ilp/connector/core/rafiki.ts`

File: packages/backend/src/payment-method/ilp/connector/core/test/mocks/accounting-service.ts [hunk 3]
```diff
@@ -133,4 +135,78 @@ export class MockAccountingService implements AccountingService {
...
     return account
   }
+
+  createDeposit(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    deposit: Deposit,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    trx?: TransactionOrKnex
+  ): Promise<void | TransferError> {
+    return Promise.resolve(undefined)
```
Comment (mkurapov) [inline packages/backend/src/payment-method/ilp/connector/core/test/mocks/accounting-service.ts:145]: We could also just throw errors for all of these to make sure they aren't used but I don't have a strong opinion

**Review by mkurapov — DISMISSED** [review]

File: packages/backend/src/payment-method/ilp/connector/core/rafiki.ts (hunk 2 above)
Comment (koekiebox) [reply packages/backend/src/payment-method/ilp/connector/core/rafiki.ts:213]: Found some more, updated. 💪

File: packages/backend/src/payment-method/ilp/connector/core/test/mocks/accounting-service.ts (hunk 3 above)
Comment (koekiebox) [reply packages/backend/src/payment-method/ilp/connector/core/test/mocks/accounting-service.ts:145]: I like that more. I updated it.

**Review by mkurapov — APPROVED** [review]

**Pull request merged**